Issue #1026

beercsaba opened the pull request:
implementation of Issue #1026

sdanyliv commented on Source/LinqToDB/DataExtensions.cs:
```diff
@@ -462,12 +503,12 @@ public static int Delete<T>([NotNull] this IDataContext dataContext, T obj)
-			string          statementHeader = null,
-			string          statementFooter = null,
-			DefaultNullable defaultNullable  = DefaultNullable.None)
+			string tableName = null,
```
Please return table formatting back. This comment is related to whole file.

beercsaba replied:
I saw this issue. my VS made this formatting :(
Do you know any way to set the tabular formatting in VS?

sdanyliv replied:
We just undo VS fromatting. And trying to disable, formatting on paste, after semicolon.
Igor made this formatting by hands to better read and understand code.

sdanyliv replied:
Before commit we ensure that we do not change non-related parts.

sdanyliv commented on Source/LinqToDB/Linq/QueryRunner.Insert.cs:
```diff
@@ -1,5 +1,4 @@
-﻿using System;
-using System.Collections.Concurrent;
+﻿using System.Collections.Concurrent;
```
Looks like this file do not need changes

beercsaba replied:
i've fixed a typo (in these QueryRunner.* classes): _queryChache should be spelled queryCache

sdanyliv commented on Source/LinqToDB/Linq/QueryRunner.InsertOrReplace.cs:
```diff
@@ -14,18 +13,23 @@ static partial class QueryRunner
 				var fieldDic = new Dictionary<SqlField,ParameterAccessor>();
 				var sqlTable = new SqlTable<T>(dataContext.MappingSchema);
+
+				if (tableName		!= null)	sqlTable.PhysicalName	= tableName;
```
Fromatting rules, are: line starts from Tab, then formatting by spaces. You can see online that formatting is broken.

sdanyliv commented:
If you doing rollbacks by hands, do’t do that i will fix everything tomorrow.
Next time be careful with such kind of commits.
Thanks!

beercsaba commented:
Yes, i'm doing it by hands.  I don't want to break your rules.
There are some small differences remained, but maybe (i hope) they are acceptable.

imho: 
from some point of view It would be better to wrap this Collection<Entity ->  (tableName, schemaName, databaseName) triplet> mapping into a CustomNaming class and make it a field of the DataContext class, which can be set in constructor. This would restrict to one-to-one mapping, but make method calls shorter (no need of these 3 parameters), so code will be easier to read. 
What do you think?
I have no time for working on this now and the next 2 months, so if you agree with my opinion, i could work on that later.

sdanyliv commented on Source/LinqToDB/DataExtensions.cs:
```diff
@@ -49,10 +47,10 @@ public static ITable<T> GetTable<T>(this IDataContext dataContext)
-			object instance,
-			[NotNull] MethodInfo methodInfo,
-			[NotNull] params object[] parameters)
+						this IDataContext	dataContext,
```
Undo formatting. Leave changed only your methods.

beercsaba replied:
done

sdanyliv commented on Source/LinqToDB/DataExtensions.cs:
```diff
@@ -98,9 +96,9 @@ public static ITable<T> GetTable<T>(this IDataContext dataContext)
 		/// <typeparam name="TDc">Type of data context parameter, passed to compiled query.</typeparam>
 		/// <typeparam name="TResult">Query result type.</typeparam>
-		public static Func<TDc,TResult> Compile<TDc,TResult>(
+		public static Func<TDc, TResult> Compile<TDc, TResult>(
```
Spaces also revert, its’ specific Igor’s style, no spaces between generic types.
For me is better to revert whole file and just copy paste your changes from previous version.

beercsaba commented:
I think i'm done with formatting, which takes almost as much time as the implementation.
i think i've fixed some formatting inconsistencies in these files. If you do not agree, i could revert these fixes.
i've renamed field _queryChache to _queryCache in there QueryRunner.... classes, i think this is a typo. If you do not agree, i could revert these too.
@sdanyliv what do you think?

sdanyliv commented:
@beercsaba, i'm picking your PR and will made corrections. Thanks for your help.

sdanyliv commented:
I have opened new PR #1045.
